src/CattyTests/DataModel/UserListTest.swift

```diff
@@ -81,11 +92,131 @@ final class UserListTest: XCTestCase {
         let userListB = UserList(name: "userListB")
         let userListC = UserList(name: "userList")
 
-        userListA.value = NSMutableArray(array: [50, 51, "item"])
-        userListB.value = NSMutableArray(array: [50, 51, "item"])
-        userListC.value = NSMutableArray(array: [50, 51, "item"])
+        userListA.add(element: 50)
+        userListB.add(element: 50)
+        userListC.add(element: 50)
+        userListA.add(element: "item")
+        userListB.add(element: "item")
+        userListC.add(element: "item")
 
         XCTAssertFalse(userListB.isEqual(userListA))
         XCTAssertTrue(userListC.isEqual(userListA))
     }
+
+    func testAdd() {
+        let list1 = UserList(name: "testName")
+
+        list1.add(element: 10)
+        list1.add(element: 20)
+
+        let value = list1.value
+
+        XCTAssertEqual(2, value.count)
+        XCTAssertEqual(value[0] as! Int, 10)
+        XCTAssertEqual(value[1] as! Int, 20)
+    }
+
+    func testDelete() {
+        let list1 = UserList(name: "testName")
+
+        list1.add(element: 10)
+        list1.add(element: 20)
+
+        var value = list1.value
+
+        XCTAssertEqual(2, value.count)
+
+        list1.delete(at: 2)
+        value = list1.value
+
+        XCTAssertEqual(1, value.count)
+        XCTAssertEqual(value[0] as! Int, 10)
+    }
+
+    func testDeleteInvalidIndex() {
+        let list1 = UserList(name: "testName")
+
+        list1.add(element: 10)
+        list1.add(element: 20)
+
+        var value = list1.value
+        XCTAssertEqual(2, value.count)
+
+        list1.delete(at: 3)
+        value = list1.value
+        XCTAssertEqual(2, value.count)
+
+        list1.delete(at: -1)
```

Comment:
Please change to lower boundary value **0** which is crucial for the implementation of the _UserList_.